core/src/main/java/tc/oc/pgm/structure/Dynamic.java

```diff
@@ -0,0 +1,69 @@
+package tc.oc.pgm.structure;
+
+import org.bukkit.util.Vector;
+import tc.oc.pgm.api.feature.Feature;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.util.BudgetWorldEdit;
+
+public class Dynamic implements Feature<DynamicDefinition> {
+
+  private final Match match;
+  final StructureDefinition structureDefinition;
+  final Vector offset;
+  private final DynamicDefinition dynamicDefinition;
+
+  // Since the passive filter can skip placing the structure,
+  // we need to keep track of whether its placed or not if we
+  // want to avoid unnecessary clears.
+  boolean placed;
+
+  Dynamic(DynamicDefinition dynamicDefinition, Match match) {
+    this.match = match;
+    this.dynamicDefinition = dynamicDefinition;
+    this.structureDefinition = this.dynamicDefinition.getStructureDefinition();
+    this.offset = this.findOffset();
```

Comment:
Given how StructureDefinition has a non-final field, i think you could also move this offset into Dynamic and lazy-load it. As long as it doesn't allow for any arbitrary change after creation (eg: no setter) and it's just a result of lazy-loading, it's fine if definitions have an *slight* amount of mutable state